libraries/AP_HAL_PX4/CAN.h

```diff
@@ -0,0 +1,332 @@
+/*
+ * Copyright (C) 2014 Pavel Kirienko <pavel.kirienko@gmail.com>
+ *
+ * With modifications for Ardupilot CAN driver
+ * Copyright (C) 2017 Eugene Shamaev
+ */
+
+#pragma once
+
+#include "AP_HAL_PX4.h"
+#include <systemlib/perf_counter.h>
+#include <AP_HAL/CAN.h>
+#include <pthread.h>
+#include <semaphore.h>
+
+#include "bxcan.h"
+#include "AP_HAL/utility/RingBuffer.h"
+
+#define UAVCAN_STM32_NUM_IFACES 2
+#define UAVCAN_STM32_RX_QUEUE_SIZE 64
+
+#ifndef UAVCAN_NULLPTR
+# if UAVCAN_CPP_VERSION >= UAVCAN_CPP11
```

Comment:
> libuavcan is built with 03 on most platforms

Just to clarify, this is because many platforms where ArduPilot is used lack a proper C++ standard library, which doesn't permit libuavcan to be compiled in C++11 mode.